Modular Parts: Asset Dump and Light Test

chryswoehler opened the pull request:
# Modular Parts: Asset Dump and Light Test
About 75ish% of floor one is laid out (critical lack of ceilings and right wing is not started). **This PR is really just an asset dump** since I'm feeling pretty good about how the modular parts are working when it comes to laying out floor1 . Also a **rough lighting test** for floor 1 (what I've set up of it so far, some rooms are pitch black lol)

## Modular parts 
- Orientation issues/origin problems fixed so it feels like a consistent pack to use when building (minimized weirdness)
- Next on my todo is finishing the **WorldAligned** shader I added, testing using world aligned textures as a way to cheat complexity via normal maps instead of manual texture detailing (deep rock inspired shader node stuff)

## Organization of modular part blocks
- Organized floor 1 parts into more neat groups. General room organization structure can be found in scene titled **"TemplateRoom"**
- Some rooms that I know will be repeated are their own grouped 'room' inside the overall 'block' grouping (concourse, security, etc) things like bathrooms, some hallways, etc.

## Lighting notes
- I'm working on issues with baked lighting, sometimes things are a *bit* off (light leaks between UV maps) So I'm messing with the auto UV margins to fix this. It can also sometimes be a tiny physical gap in game i missed casting light lol.
- Set modular parts as static objects to test emissive. Let me know what you guys think of the low-light emergency backup look I've set up in the first few rooms. 
- Baking averages like 10 minutes or so for the floor as it is now, which could be better honestly so I'll be doing some setting tweaks as I keep working.

## Scaling of rooms/Player/Creature
- Please let me know how things feel in scene, I have the creature at his rough height for in game (maybe a little tall). The player got sized down since my last push but still may need more.

Jodiiiiiiii reviewed (approved):
# **Modular Parts: Asset Dump & Light Test - PR Review**
Overall great stuff, the only things I think should be addressed before merge are simple Assets folder structure stuff. Otherwise I've just elaborated on some of my thoughts/feedback/ideas below!
## Stuff to Address Before Merge
- I see you made a new **"Materials" folder** for some stuff you were using - we have a "Shaders & Materials" folder lower down if you want to make a subfolder in there or something. Or if we all want we can split shaders and materials into separate Assets folders but I think we should pick one of these options sooner than later
- You also made a new **"Shaders" folder** - same note as above
## Modular Prefabs - Assets
- lots of good models & prefabs!
- **Prefabs not marked as static**: DoorPlainWall, Elevator (maybe intentional?? unsure), HangarDoor, FullWindow, HalfWindow,  MidWindow
- security short walls also aren't static (in Floor1 scene), but its cause they are using the model instead of the prefab (considering they are placeholders this is probably inconsequential but it may be a large contributing factor to the bleeding seen here since they aren't static?)
## Modular Organization Structure
- looks great! I appreciate the clear structure to it
- I just had the thought that your life may be made easier with **Grid Snapping** (must set coord mode to global)
  - this might require more fiddling to make all the prefabs align the same with the grid snapping but its up to you if its worth it - I think it would probably save time in the long run
  - also grid snapping is based on global coordinates only so it should work even with arbitrarily placed empty containers. so if you want to align empties cleanly as well then I believe you can grid snap the whole thing around to adjust entire room placements with snapping on the empty (would require some reconfiguring up front but again - up to you)
  - this might *potentially* help with the light bleeding issue but I am not certain
## Floor1 Scene
- a couple walls are in the Cargo empty when they should be in Engineering based on grouping of surrounding walls
- I thought we had floor 1 bathrooms off concourse, not the side room/hallways, but it honestly doesn't matter much since its not critical to progression
- I fell through the curved bit of the railway station lol
## Lighting
- the ambient light rings on hallways are immaculate, I love it
- Baking times: I feel like 10 minutes sounds normal for baking our lights, and doing it periodically or mainly just near the end of development once most static props are placed should be good I imagine - thankfully our realtime lights *shouldn't* impact this at all
- I see you have turned on "Light Probes: Blend Probes" on certain player model objects and on the testing terminal. There's these large circle things around the models and I don't actually know what they do - but I think they are supposed to allow realtime objects to receive baked light in some capacity? somewhat unsure but did you figure something out here or should those be disabled? (especially the terminal probably shouldn't have them since terminals don't move)
## Player/Creature Scaling
- Player scaling feels good visually - could maybe be a bit smaller/lower? (the doors look so tiny in the distance lol)
  - there seems to be a place where the player actually gets stuck on doors, and I don't think its hitting its head but rather getting stuck on the bottom lip of the door circle lip thing? Maybe we can tone back the physical collider on the door to not exactly match the door?
  - when stuck on door, you literally cannot move, and when I jumped I got wedged in the top of the door and was stuck hanging there
  - jumping in main room actually clips camera through the ceiling (no ceiling collider? or we could decrease the jump height which we may do anyways?)
- Creature seems pretty solid, but I lowkey think we could make it even bigger if we want (as long as it fits vertically). If the animation tucks the draping bottom section more horizontally rather than hanging it would also give more space, now sure if that was gonna be done in animation at all
  - a different idea is that the creature could be shorter when in idle and it literally raises vertically when in pursuit with the tail piece hanging down slightly more than when in idle

chryswoehler commented:
# **Modular Parts: Asset Dump & Light Test - PR Review, Updated PR**
Fixes to address PR review notes. Also some more floor 1 progress updates.

## Stuff to Addressed Before Merge
- Completely missed the "Shaders & Materials" folder, I merged main in last second before pushing so a bit better review could have been done on my part. I went ahead and put the **"Materials" folder** and **"Shaders" folder** into **"Shaders & Materials"** as sub folders. I think I will end up with a lot of individual materials the way I am not approaching textures, so keeping them in sub folders will make things less messy as we go on.
## Modular Prefabs - Assets
- **Prefabs not marked as static**: 
 - *DoorPlainWall*: Missed, fixed.
 - *Elevator, HangarDoor*: both temp sizing models, just haven't needed to set up fully
 - *FullWindow, HalfWindow,  MidWindow*: missed, fixed.
 - *security short walls aren't static*: temp stuff also

## Modular Organization Structure
- **Grid Snapping**
 1. Some tips if either of you need to adjust building stuff later on: 
 - For whole object 'steps' a modular panel is 3 units wide/tall in unity. half wall variants are 1.5 wide by 3 tall. You can either use those values while snapping or just +/- in the translate sections.
   - For general moving a step of 0.5 works good
   - For fine moving I use 0.1 or 0.05 to get things settled against eachother and then 0.01 if I can see the tiniest crack still showing through

## Floor1 Scene
- *"walls in the Cargo empty when they should be in Engineering based on grouping of surrounding walls"* Fixed!
- *"floor 1 bathrooms off concourse, not the side room/hallways"* At least on the last progress map last sent it was on hallway, I might be one out of date? But easy to fix and like you said not hugely impactful either way.
- *"I fell through the curved bit of the railway station"* I had a collider but it was just a bit too complex, added a simplified version of the mesh to work as a mesh collider so it's fixed.

## Lighting
- Baking times: "10 minutes sounds normal" Cool! Wont get too into fiddling with it for now then.
- *"Light Probes: Blend Probes"* I never noticed these? I think it may be some sort of hold over from the original playground scene Andrew set up, clearing them out doesn't seem to hurt anything so it got pruned for now.
- *"light bleeding - does it happen anywhere outside of security*" Not often only a few known bleed causing issues to avoid:
   - using some of the columns outside map made the walls 'glow' even when no physical crack for light to come through. Just turned off their light bar objects since it isn't seen anyway
   - Some of the earliest walls placed will have lighting issues? I'm still hunting these down as they pop up but I think something just got off about the prefab

## Player/Creature Scaling
- *"there seems to be a place where the player actually gets stuck on doors"*+ *"tone back the physical collider on the door to not exactly match the door*" Trying something new with some box colliders. I think this is better now.

## World Aligned Shader Material System
- Made a shader node system for texturing objects using world space aligned texturing.
- Any tiling texture can be swapped into the base material slots
- Variables for roughness, base and noise independently
- Variables for metallics, base and noise independently
- Variables for overall noise sizing, smoothness

##Asset Dump
- Models and prefabs created for all of the 20 new decor models
- Still missing colliders on each, to be added in next PR with full floor 1 layout

Jodiiiiiiii reviewed (approved):
# **Modular Parts: Asset Dump & Light Test - PR Re-Review**

I am gonna go ahead and merge it!! It all looks super great with the HUD in now too! Good job with the wall/floor additions for the whole floor that will be nice.

## Things I did
- I marked all the new prop/decor prefabs as static for lighting purposes since I figured that should be done. feel free to undo it on your next branch if you had a reason for not doing so - I figured it was a harmless quick fix if this was unintentional

## World-Aligned Shader
- Is it actually working in your other scene? fiddling in play mode didn't seem to do anything. Is it WIP or otherwise not actually what I thought world-aligned textures were?

## Grid Snapping
By bringing up grid snapping I sort of meant that I think there is a way to do it entirely with grid snapping (meaning no fine tuning adjustments like you described - although that will be inevitable for props. Regardless, if you have a system down and it works then that's all good by me!
